modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/filters/modal_content/DateRangeFilter.tsx

```diff
@@ -5,102 +5,243 @@
 
 import ClayDatePicker from '@clayui/date-picker';
 import ClayForm from '@clayui/form';
+import ClayModal from '@clayui/modal';
 import classNames from 'classnames';
 import {format, getYear, isBefore, isEqual} from 'date-fns';
-import React, {useEffect} from 'react';
+import React, {useState} from 'react';
+
+import FilterModalConfiguration from '../../../components/FilterModalConfiguration';
+import FilterModalFooter from '../../../components/FilterModalFooter';
+import {IDateFilter, IField, IFilter} from '../../../utils/types';
 
 function Header() {
 	return <>{Liferay.Language.get('new-date-range-filter')}</>;
 }
 
 interface IBodyProps {
-	from: string;
-	isValidDateRange: boolean;
+	closeModal: Function;
+	fieldNames?: string[];
+	fields: IField[];
+	filter?: IFilter;
+	handleSave: Function;
 	namespace: string;
-	onFromChange: (val: string) => void;
-	onToChange: (val: string) => void;
-	onValidDateChange: (val: boolean) => void;
-	to: string;
 }
 
 function Body({
-	from,
-	isValidDateRange,
+	closeModal,
+	fieldNames,
+	fields,
+	filter,
+	handleSave,
 	namespace,
-	onFromChange,
-	onToChange,
-	onValidDateChange,
-	to,
 }: IBodyProps) {
+	const [fieldInUseValidationError, setFieldInUseValidationError] = useState<
+		boolean
+	>(false);
+	const [saveButtonDisabled, setSaveButtonDisabled] = useState<boolean>(
+		filter ? false : true
+	);
+	const fdsFilterLabelTranslations = filter?.label_i18n ?? {};
+	const [i18nFilterLabels, setI18nFilterLabels] = useState(
+		fdsFilterLabelTranslations
+	);
+
+	const inUseFields: (string | undefined)[] = fields.map((item) =>
+		fieldNames?.includes(item.name) ? item.name : undefined
+	);
+
+	const [selectedField, setSelectedField] = useState<IField | undefined>(
+		fields.find((item) => item.name === filter?.fieldName)
+	);
+	const [from, setFrom] = useState<string>(
+		(filter as IDateFilter)?.from ?? ''
+	);
+	const [to, setTo] = useState<string>((filter as IDateFilter)?.to ?? '');
+	const [isValidDateRange, setIsValidDateRange] = useState<boolean>(true);
+
 	const fromFormElementId = `${namespace}From`;
 	const toFormElementId = `${namespace}To`;
 
-	useEffect(() => {
-		let isValid = true;
+	const handleFilterSave = () => {
+		const body = {
+			fieldName: selectedField?.name,
+			from,
+			label_i18n: i18nFilterLabels,
+			to,
+			type: selectedField?.format,
+		};
+
+		handleSave(body);
+	};
+
+	const isFormInvalid = ({
+		from,
+		i18nFilterLabels,
+		selectedField,
+		to,
+	}: {
+		from: string;
+		i18nFilterLabels: Partial<Liferay.Language.FullyLocalizedValue<string>>;
+		selectedField: IField | undefined;
+		to: string;
+	}) => {
+		if (!selectedField) {
+			return true;
+		}
+
+		if (selectedField && !filter) {
+			if (inUseFields.includes(selectedField.name)) {
+				return true;
+			}
+		}
+
+		if (!i18nFilterLabels || !Object.values(i18nFilterLabels).length) {
+			return true;
+		}
+		else {
```

Comment:
Why are we validating `i18nFilterLabels` and `selectedField` here, and not in the common section component? We are setting `valid` in callback, so all of this can be hidden.